homeassistant/components/mqtt/client.py

```diff
@@ -671,16 +671,32 @@ def _mqtt_on_connect(
             )
             return
 
-        self.connected = True
-        dispatcher_send(self.hass, MQTT_CONNECTED)
         _LOGGER.info(
             "Connected to MQTT server %s:%s (%s)",
             self.conf[CONF_BROKER],
             self.conf[CONF_PORT],
             result_code,
         )
 
-        self.hass.create_task(self._async_resubscribe())
+        self.hass.create_task(self._async_connect())
+
+    async def _async_connect(self) -> None:
```

Comment:
Please don't move code around like this, it makes reviewing harder. The moving around can be done in a separate PR instead.

Reply:
I meant that you have renamed `_async_resubscribe` to `_async_connect` and moved it around in the class

Reply:
@emontnemery Again - at @bdraco 's suggestion.  Explicitly.  And no, I didn't just rename it.
https://github.com/home-assistant/core/pull/88072#discussion_r1109153728

_async_resubscribe was only EVER being invoked from on_connect.  And by creating _async_connect, I actually took a significant chunk of on_connect, the chunk that can/should be done async anyway, and moved it to that function, combining it with the existing async_resubscribe code (which was only being invoked by that function anyway).